style_guide/README.md

```diff
@@ -0,0 +1,200 @@
+    you and the reviewer(s) are satisfied.
+
+11. Once the pull request has been accepted, you can merge the pull request and
+    delete the feature branch.
+
+
+### Making hotfixes
+
+As shown in the diagram above, sometimes hotfixes need to be directly to the
+`main` branch. The workflow for this scenario is as follows:
+
+1. Assuming steps (1) through (3) in the previous section are already completed,
+   create a new branch named `hotfix-<description>` off of the `main` branch,
+   and commit any necessary changes following the nominal `git add`/`git commit`
+   cycle.
+2. Push the hotfix branch to your personal fork, and open two separate pull
+   requests: one that merges the hotfix branch into the
+   `IMAP-Science-Operations-Center` `imap_processing` `main` branch, and one
+   that merges the hotfix branch into the `IMAP-Science-Operations-Center`
+   `imap_processing` `dev` branch.
+3. For each of these pull requests, proceed with the nominal review & merge
+   process described in steps (9) and (10) in the previous section.
+
+
+### Keeping your fork updated
+
+You can keep your personal fork up-to-date with the
+`IMAP-Science-Operations-Cetner` `imap_processing` repository by fetching and
+pulling the `upstream` remote:
+
+    git checkout dev
+    git fetch upstream dev
+    git pull upstream/dev
+
+### Collaborating on someone else's fork
+
+To contribute to a branch on another person's personal fork, add a new `remote`
+that points to their fork, and use the nominal workflow for contributing:
+
+    git remote add <username> <remote url>
+    git fetch <username>
+    git checkout -b <branchname> <username>/<branchname>
+    # Make some changes via add/commit cycle
+    git push <username> <branchname>
+
+Python Coding
+-------------
+
+`imap_processing` code shall adhere to the [PEP8](https://www.python.org/dev/peps/pep-0008/)
+conventions save for the following exceptions:
+
+ - Lines of code need not be restricted to 79 characters.  However, it is
+encouraged to break up excessively long lines into several lines if it benefits
+the overall readability of the code
+
+ Additionally, the code shall adhere to the following special guidelines:
+
+ - Function and class definitions should be placed in alphabetical order in the
+module
+ - It is encouraged to annotate variables and functions using the
+[`typing`](https://docs.python.org/3/library/typing.html) library (see
+[PEP 483](https://www.python.org/dev/peps/pep-0483/),
+[PEP 484](https://www.python.org/dev/peps/pep-0484/), and
+[PEP 526](https://www.python.org/dev/peps/pep-0526/)).
+
+
+Documentation
+-------------
+
+`imap_processing` code shall adhere to the [PEP257](https://www.python.org/dev/peps/pep-0008/)
+and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) conventions.
+
+The following are further recommendations:
+
+- Each module should have at minimum a description, `Authors`, and `Use`
```

Comment:
I'm not sure I personally like `Authors` sections... I use `git blame` if I need to see who made changes to individual areas. I like `IMAP SOC Contributors` or something more generic to be an inclusive feeling project. I could see a `contributors.md` list or something if people want names associated somewhere though.

Reply:
Makes sense.  I like the idea of keeping track of who has contributed to the project, either by direct code contributions (which GitHub already keeps track for you) but also for anyone who provided some sort of expertise or help otherwise.  I'm not sure if that list would quickly get out of hand though?  I kept an 'Acknowledgements' list in a [previous project](https://github.com/spacetelescope/jwql#acknowledgments) and it did get a bit overwhelming, but it made it really easy to thank people in presentations/publications. 🤷🏻

Reply:
There are a few projects I've contributed to that add you to the list of contributors and put info on a markdown page somewhere. For example here: https://github.com/pyproj4/pyproj#contributors-

I also generally use the "release notes" feature in GitHub to automatically generate the release notes and that indicates who contributed what (PR link + author) (a project with only me as the contributor, but you get the idea: https://github.com/SWxTREC/pymsis/releases/tag/v0.7.0 )

I would be in favor of either of these approaches over the per-module approach.

Reply:
The first example you shared looks really cool, we should try it!  I opened #22 for it.